pkg/util/util.go

```diff
@@ -61,6 +63,53 @@ func loadCertsFromPaths(paths []string, pool *x509.CertPool) (*x509.CertPool, er
 	return pool, nil
 }
 
+func getClientCertificates(certFile, keyFile string) ([]tls.Certificate, error) {
+	if _, err := os.Stat(certFile); os.IsNotExist(err) {
+		return nil, fmt.Errorf("unable to locate certificate %s: %w", certFile, err)
+	}
+
+	if _, err := os.Stat(keyFile); os.IsNotExist(err) {
+		return nil, fmt.Errorf("unable to locate key %s: %w", keyFile, err)
+	}
+
+	cert, err := tls.LoadX509KeyPair(certFile, keyFile)
+	if err != nil {
+		return nil, fmt.Errorf("could not parse cert/key pair %s, %s: %w", certFile, keyFile, err)
+	}
+
+	return []tls.Certificate{cert}, nil
+}
+
+func GetHTTPClient(certFile, keyFile string, insecureSkipVerify bool, useSystemPool bool, caFiles ...string) (*http.Client, error) {
+	transport := http.DefaultTransport.(*http.Transport).Clone()
+	transport.TLSClientConfig = &tls.Config{
+		MinVersion: tls.VersionTLS12,
```

Comment:
Needs to be mentioned in the documentation

Reply:
I think TLS 1.2 was already the [default](https://github.com/ethanchowell/oauth2-proxy/blob/e3dc927e570700ae598425344cf07073965bd19c/pkg/validation/options.go#L45) but happy to add that to the docs as well.